From 80df8d73037bb5f3a4a662299ca14389afb8e610 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:24 -0400
Subject: [net] Allow csum_add to be provided in arch

Message-id: <1409652920-431-2-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90137
O-Subject: [PATCH 01/57] [RHEL7.1 net] net: Allow csum_add to be provided in arch
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 07064c6e022ba8dc0c86ce12f7851a1de24e04fc
Author: Tom Herbert <therbert@google.com>
Date:   Fri May 2 16:28:03 2014 -0700

    net: Allow csum_add to be provided in arch

    csum_add is really nothing more then add-with-carry which
    can be implemented efficiently in some architectures.
    Allow architecture to define this protected by HAVE_ARCH_CSUM_ADD.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/checksum.h b/include/net/checksum.h
index fd4c48d..7730407 100644
--- a/include/net/checksum.h
+++ b/include/net/checksum.h
@@ -57,12 +57,14 @@ static __inline__ __wsum csum_and_copy_to_user
 }
 #endif
 
+#ifndef HAVE_ARCH_CSUM_ADD
 static inline __wsum csum_add(__wsum csum, __wsum addend)
 {
  u32 res = (__force u32)csum;
  res += (__force u32)addend;
  return (__force __wsum)(res + (res < (__force u32)addend));
 }
+#endif
 
 static inline __wsum csum_sub(__wsum csum, __wsum addend)
 {
-- 
1.7.1